### DISCUSSION TITLE
Would it be possible to run Waydroid on a MacBook (M1) ?

### DISCUSSION THREAD

**Discussion by deslunes** [waydroid/waydroid#942] [General]
Hello! All I'm saying may be wrong anti may not in get your interest but I'm looking for a way to get android apps running on Mac. While doing some research I discovered waydroid, and since macOS has a pretty similar terminal as linux I was wondering if wayDroid would work or run on Macs. What do you think about that?

**Comment by SuperKenVery** [comment]
Waydroid requires Linux namespaces which are not present on MacOS. Asahi Linux is promising, but we still need to wait for Vulkan drivers and a 4k page size kernel.

**Reply by glowinthedark** [reply]
How about running on silicon mac under UTM with an Ubuntu arm64 image running wayland? My attempts to do this under a guest running ubuntu-22.04.3 arm64 were unsuccessful because couldn't get binder to work — is that because of known limitations imposed by QEMU or some misconfiguration? I did make the adjustments listed in https://docs.waydro.id/faq/get-waydroid-to-work-through-a-vm although the page does not list any specifics by OS/vm-type. Looks like most issues people are having while trying to run waydroid are related to specific missing kernel modules, such as `binder` and `ashmem` and going through a bunch of github issues so far didn't help find the specific steps required for arm64 macs.

**Update**: It is possible to get waydroid up and running on macos ARM64 under UTM under [ubuntu arm64](https://cdimage.ubuntu.com/ubuntu/releases/22.04.3/) v22.04.3 guest with the workaround described here: https://github.com/waydroid/waydroid/issues/644#issuecomment-1677922061 with [custom images](https://drive.google.com/file/d/1Ikk7BnwAGFT28LHMMTVQjbGdPBQz6nne/view?pli=1) installed according to https://docs.waydro.id/faq/using-custom-waydroid-images

**Reply by dharmtej-0206** [reply]
Have you tried genymotion?

**Comment by SuperKenVery** [comment]
Now that Vulkan drivers and 4k VM are all present, I think it should be possible to run waydroid on M1 with asahi!
